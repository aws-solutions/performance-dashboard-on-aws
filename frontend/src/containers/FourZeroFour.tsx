/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import Link from "../components/Link";
import Footer from "../layouts/Footer";
import { useTranslation } from "react-i18next";

function FourZeroFour() {
    const { t } = useTranslation();

    return (
        <>
            <div className="text-center">
                <p className="font-sans-3xl text-heavy margin-top-9 margin-bottom-1">
                    {t("PageNotFound")}
                </p>
                <hr className="width-tablet border-base-light" />
                <p className="font-sans-md">
                    {t("PageNotFoundDescription")} <Link to="/">homepage</Link>.
                </p>
            </div>
            <Footer />
        </>
    );
}

export default FourZeroFour;
