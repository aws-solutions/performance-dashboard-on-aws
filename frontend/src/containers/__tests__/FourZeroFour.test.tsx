/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import FourZeroFour from "../FourZeroFour";

test("renders a FourZeroFour component", async () => {
    const wrapper = render(<FourZeroFour />, { wrapper: MemoryRouter });
    expect(wrapper.container).toMatchSnapshot();
});
