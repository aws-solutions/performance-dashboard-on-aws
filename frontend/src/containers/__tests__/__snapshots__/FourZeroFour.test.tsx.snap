// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a FourZeroFour component 1`] = `
<div>
  <div
    class="text-center"
  >
    <p
      class="font-sans-3xl text-heavy margin-top-9 margin-bottom-1"
    >
      404 / Page not found
    </p>
    <hr
      class="width-tablet border-base-light"
    />
    <p
      class="font-sans-md"
    >
      You may want to double-check your link and try again, or return to the
       
      <a
        aria-label=""
        class="link "
        href="/"
        target=""
      >
        <span>
          homepage
        </span>
      </a>
      .
    </p>
  </div>
</div>
`;
